src/oneD/StFlow.cpp

```diff
@@ -17,1265 +12,303 @@ namespace Cantera
 {
 
 StFlow::StFlow(ThermoPhase* ph, size_t nsp, size_t points) :
-    Domain1D(nsp+c_offset_Y, points),
-    m_nsp(nsp)
+    Flow1D(ph, nsp, points)
 {
-    m_points = points;
-
-    if (ph == 0) {
-        return; // used to create a dummy object
-    }
-    m_thermo = ph;
-
-    size_t nsp2 = m_thermo->nSpecies();
-    if (nsp2 != m_nsp) {
-        m_nsp = nsp2;
-        Domain1D::resize(m_nsp+c_offset_Y, points);
-    }
-
-    // make a local copy of the species molecular weight vector
-    m_wt = m_thermo->molecularWeights();
-
-    // set pressure based on associated thermo object
-    setPressure(m_thermo->pressure());
-
-    // the species mass fractions are the last components in the solution
-    // vector, so the total number of components is the number of species
-    // plus the offset of the first mass fraction.
-    m_nv = c_offset_Y + m_nsp;
-
-    // enable all species equations by default
-    m_do_species.resize(m_nsp, true);
-
-    // but turn off the energy equation at all points
-    m_do_energy.resize(m_points,false);
-
-    m_diff.resize(m_nsp*m_points);
-    m_multidiff.resize(m_nsp*m_nsp*m_points);
-    m_flux.resize(m_nsp,m_points);
-    m_wdot.resize(m_nsp,m_points, 0.0);
-    m_hk.resize(m_nsp, m_points, 0.0);
-    m_dhk_dz.resize(m_nsp, m_points - 1, 0.0);
-    m_ybar.resize(m_nsp);
-    m_qdotRadiation.resize(m_points, 0.0);
-
-    //-------------- default solution bounds --------------------
-    setBounds(c_offset_U, -1e20, 1e20); // no bounds on u
-    setBounds(c_offset_V, -1e20, 1e20); // no bounds on V
-    setBounds(c_offset_T, 200.0, 2*m_thermo->maxTemp()); // temperature bounds
-    setBounds(c_offset_L, -1e20, 1e20); // lambda should be negative
-    setBounds(c_offset_E, -1e20, 1e20); // no bounds on electric field
-    setBounds(c_offset_Uo, -1e20, 1e20); // no bounds on Uo
-    // mass fraction bounds
-    for (size_t k = 0; k < m_nsp; k++) {
-        setBounds(c_offset_Y+k, -1.0e-7, 1.0e5);
-    }
-
-    //-------------------- grid refinement -------------------------
-    m_refiner->setActive(c_offset_U, false);
-    m_refiner->setActive(c_offset_V, false);
-    m_refiner->setActive(c_offset_T, false);
-    m_refiner->setActive(c_offset_L, false);
-    m_refiner->setActive(c_offset_Uo, false);
-
-    vector<double> gr;
-    for (size_t ng = 0; ng < m_points; ng++) {
-        gr.push_back(1.0*ng/m_points);
-    }
-    setupGrid(m_points, gr.data());
-
-    // Find indices for radiating species
-    m_kRadiating.resize(2, npos);
-    m_kRadiating[0] = m_thermo->speciesIndex("CO2");
-    m_kRadiating[1] = m_thermo->speciesIndex("H2O");
+    warn_deprecated("StFlow::StFlow",
+        "To be removed after Cantera 3.0. Class replaced by Flow1D.");
 }
```

Comment:
"after Cantera **3.1**", here and in several other places.

Reply:
Wow, that decided to show a lot of diff lines...

Reply:
Good catch!